root/usr/share/fdi/facts/ethtool.rb

```diff
@@ -4,7 +4,7 @@
 # See: https://community.theforeman.org/t/foreman-discovery-image-3-7/21642
 #
 # To test:
-# $ FACTERLIB=/usr/share/fdi/facts tfm-ruby /opt/theforeman/tfm/root/usr/bin/facter
+# $ FACTERLIB=/usr/share/fdi/facts ruby /opt/theforeman/tfm/root/usr/bin/facter
 
 require 'facter'
```

Comment:
On line 24 you have `Facter.value(:networking)['interfaces']` but using `Facter.value('networking.interfaces')` is more efficient.

You're also using `Facter::Core::Execution.exec` which is deprecated in favor of `.execute`: https://www.rubydoc.info/gems/facter/Facter/Core/Execution#exec-class_method

Reply:
I would not like to do any changes other than required to get this beast working on EL8.